### PR TITLE
Add Log Analytics support for ACI

### PR DESCRIPTION
**Pull request by metacpp**
This PR is mainly to addressing the feature request in #2755, which includes:

1. `log_analytics` field support in ACI.
2.  Fix the async issue in ACI CreateOrUpdate method.

### CODE REVIEW

**Comment by metacpp** [comment]
All tests passed:
image

File: azurerm/resource_arm_container_group.go [hunk 1]
```diff
@@ -279,6 +279,48 @@ func resourceArmContainerGroup() *schema.Resource {
 				},
 			},
 
+			"log_analytics": {
+				Type:     schema.TypeSet,
```
Comment (katbyte) [inline azurerm/resource_arm_container_group.go]: As this is a single block, this should be a `TypeList`

File: azurerm/resource_arm_container_group.go [hunk 2]
```diff
@@ -279,6 +279,48 @@ func resourceArmContainerGroup() *schema.Resource {
...
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"workspace_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.NoEmptyStrings,
```
Comment (katbyte) [inline azurerm/resource_arm_container_group.go]: We should be using `azure.ValidateResourceID` here

File: azurerm/resource_arm_container_group.go [hunk 3]
```diff
@@ -647,6 +705,33 @@ func expandContainerVolumes(input interface{}) (*[]containerinstance.VolumeMount
...
+	}
+
+	log := logList[0].(map[string]interface{})
+	ws_id := log["workspace_id"].(string)
+	ws_key := log["workspace_key"].(string)
+	log_type := log["log_type"].(string)
+	metadataMap := log["metadata"].(map[string]interface{})
+	metadata := make(map[string]*string)
+	for k, v := range metadataMap {
+		// TODO:
```
Comment (katbyte) [inline azurerm/resource_arm_container_group.go]: empty todp?

File: azurerm/resource_arm_container_group_test.go [hunk 4]
```diff
@@ -217,6 +217,7 @@ func TestAccAzureRMContainerGroup_linuxComplete(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "container.0.volume.0.read_only", "false"),
 					resource.TestCheckResourceAttr(resourceName, "os_type", "Linux"),
 					resource.TestCheckResourceAttr(resourceName, "restart_policy", "OnFailure"),
+					resource.TestCheckResourceAttr(resourceName, "log_analytics.#", "1"),
 				),
```
Comment (katbyte) [inline azurerm/resource_arm_container_group_test.go]: Could we check some more values here? metadata/workspace_id etc

File: azurerm/resource_arm_container_group_test.go [hunk 5]
```diff
@@ -293,6 +294,7 @@ func TestAccAzureRMContainerGroup_windowsComplete(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "container.0.secure_environment_variables.secureFoo1", "secureBar1"),
 					resource.TestCheckResourceAttr(resourceName, "os_type", "Windows"),
 					resource.TestCheckResourceAttr(resourceName, "restart_policy", "Never"),
+					resource.TestCheckResourceAttr(resourceName, "log_analytics.#", "1"),
```
Comment (katbyte) [inline azurerm/resource_arm_container_group_test.go]: Could we check some more values here? metadata/workspace_id etc

File: azurerm/resource_arm_container_group.go [hunk 6]
```diff
@@ -279,6 +279,48 @@ func resourceArmContainerGroup() *schema.Resource {
...
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+						"log_type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(containerinstance.ContainerInsights),
+								string(containerinstance.ContainerInstanceLogs),
+							}, true),
```
Comment (JunyiYi) [inline azurerm/resource_arm_container_group.go]: Could we use case-sensitive comparison for enumeration values? This is a new pattern (for your reference: https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_dev_test_linux_virtual_machine.go#L66) to deal with enumerations.
```suggestion
							}, false),
```

File: azurerm/resource_arm_container_group_test.go (hunk 4 above)
Comment (metacpp) [reply azurerm/resource_arm_container_group_test.go]: It was `set`, that's why i am skipping it. Now, it's changed to list, i will add more validations.

**Comment by metacpp** [comment]
@katbyte @JunyiYi Just updated PR, please review the changes.

**Comment by metacpp** [comment]
All ACI tests passed:
image

**Comment by tombuildsstuff** [comment]
Tests pass:

screenshot 2019-03-07 at 08 41 16

**Pull request merged**